flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/TaskMailboxExecutorService.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.tasks.mailbox;
+
+import javax.annotation.Nonnull;
+
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Interface for an {@link ExecutorService} build around a {@link Mailbox}-based execution model.
+ */
+public interface TaskMailboxExecutorService extends TaskMailboxExecutor, ExecutorService {
+
+	/**
+	 * Returns the mailbox that manages the execution order.
+	 *
+	 * @return the mailbox.
+	 */
+	@Nonnull
+	Mailbox getMailbox();
```

Comment:
Why here needs to expose the mailbox ?  Can we delegate all the interactions with mailbox to the MailboxExecutorService?